Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs

```diff
@@ -669,8 +767,25 @@ computeActualDistance orgId pickup driverPoolResults = do
           actualDurationToPickup = distDur.duration,
           intelligentScores = IntelligentScores Nothing Nothing Nothing Nothing Nothing defaultPopupDelay,
           isPartOfIntelligentPool = False,
-          keepHiddenForSeconds = Seconds 0
+          keepHiddenForSeconds = Seconds 0,
+          goHomeReqId = Nothing
+        }
+
+refactorRoutesResp :: GoHomeConfig -> (QP.NearestGoHomeDriversResult, Maps.RouteInfo, Id DDGR.DriverGoHomeRequest) -> (QP.NearestGoHomeDriversResult, Maps.RouteInfo, Id DDGR.DriverGoHomeRequest)
+refactorRoutesResp goHomeCfg (nearestDriverRes, route, ghrId) = (nearestDriverRes, newRoute route, ghrId)
+  where
+    newRoute route' =
+      RouteInfo
+        { distance = route'.distance,
+          duration = route'.duration,
+          points = tail $ refactor [] route'.points,
+          snappedWaypoints = route'.snappedWaypoints,
+          boundingBox = Nothing
         }
+    refactor acc (p1 : p2 : ps) = if getDistanceBetweenCoords p1 p2 > goHomeCfg.goHomeWayPointRadiusRadius then refactor (p1 : acc) (getPointInBetween p1 p2 (fromIntegral (goHomeCfg.goHomeWayPointRadiusRadius.getMeters)) : p2 : ps) else refactor (p1 : acc) (p2 : ps)
```

Comment:
Co-pilot

Reply:
suggestion:

```
refactor acc (p1 : p2 : ps) 
    | getDistanceBetweenCoords p1 p2 > goHomeCfg.goHomeWayPointRadiusRadius = refactor (p1 : acc) (getPointInBetween p1 p2 (fromIntegral (goHomeCfg.goHomeWayPointRadiusRadius.getMeters)) : p2 : ps)
    | otherwise = refactor (p1 : acc) (p2 : ps)
refactor acc [p1] = reverse (p1 : acc)
refactor _ [] = []
```